### PR TITLE
alphaNormalize

### PR DESCRIPTION
**Pull request by davidmanzanares**
Fix https://github.com/CartoDB/carto-vl/issues/18

Example: `examples/editor/index.html#eyJhIjoibW5tYXBwbHV0byIsImIiOiIiLCJjIjoiZG1hbnphbmFyZXMiLCJkIjoiaHR0cHM6Ly97dXNlcn0uY2FydG8uY29tIiwiZSI6ImNvbG9yOiBhbHBoYU5vcm1hbGl6ZShyYW1wKGxpbmVhcigkYXNzZXNzbGFuZCwgMCwgZ2xvYmFsUGVyY2VudGlsZSgkYXNzZXNzbGFuZCwgOTgpKSwgdGVtcHMpLCAkbnVtZmxvb3JzKVxuc3Ryb2tlQ29sb3I6IG9wYWNpdHkod2hpdGUsIDAuKSIsImYiOnsibG5nIjotNzMuOTQ1NDQxMzQwNjIwMjgsImxhdCI6NDAuNzQ2MjQyOTg1ODQ1NDd9LCJnIjoxMi40MzI1OTAxMTI0ODcyNjUsImgiOiJEYXJrTWF0dGVyIiwiaSI6ImRhdGFzZXQifQ==`

It also addresses https://github.com/CartoDB/carto-vl/issues/878 for binary operations because it made was a requirement for this.

### CODE REVIEW

File: src/renderer/viz/expressions/binary.js [hunk 1]
```diff
@@ -455,11 +455,6 @@ function genBinaryOp (name, allowedSignature, jsFn, glsl) {
             b = implicitCast(b);
 
             const signature = getSignatureLoose(a, b);
-            if (signature !== undefined) {
```
Comment (Jesus89) [inline src/renderer/viz/expressions/binary.js]: Why is this removed?

Comment (davidmanzanares) [reply src/renderer/viz/expressions/binary.js]: Because it was a psuedo `checkLooseType` check, see https://github.com/CartoDB/carto-vl/issues/878

File: test/unit/renderer/viz/expressions/binary.test.js [hunk 2]
```diff
@@ -8,46 +8,46 @@ s.FALSE.toString = () => 's.FALSE';
 describe('src/renderer/viz/expressions/binary', () => {
     describe('error control', () => {
         describe('Signature NUMBERS_TO_NUMBER | NUMBER_AND_COLOR_TO_COLOR | COLORS_TO_COLOR', () => {
-            validateDynamicTypeErrors('mul', ['number', 'category']);
-            validateDynamicTypeErrors('mul', ['category', 'number']);
-            validateDynamicTypeErrors('mul', ['category', 'category']);
+            validateCompileTypeError('mul', ['number', 'category']);
```
[the remark this answers is not in the document]
Comment (davidmanzanares) [reply test/unit/renderer/viz/expressions/binary.test.js]: Because of the removal of the extra check on `binary.js`, these tests checked the existence of pseudo `checkLooseType` checks, but I removed those, see Jesus's comment
